packages/headless/src/features/insight-user-actions/insight-user-actions-state.ts

```diff
@@ -15,15 +16,15 @@ export interface UserSession {
 export interface UserAction {
   actionType: UserActionType;
   timestamp: string;
-  eventData: {
-    type: string;
+  eventData?: {
+    type?: string;
     value?: string;
   };
   cause?: string;
   searchHub?: string;
   document?: {
-    title: string;
-    clickUri: string;
+    title?: string;
+    clickUri?: string;
```

Comment:
Is it optional? 

If you have a document, you should have a title at the very least.